kornia/geometry/boxes.py

```diff
@@ -174,37 +271,17 @@ def from_tensor(cls, boxes: torch.Tensor, mode: str = "xyxy", validate_boxes: bo
                      [7., 3.],
                      [5., 3.]]])
         """
-        if not (2 <= boxes.ndim <= 3 and boxes.shape[-1] == 4):
-            raise ValueError(f"Boxes shape must be (N, 4) or (B, N, 4). Got {boxes.shape}.")
-
-        batched = boxes.ndim == 3
-        boxes = boxes if batched else boxes.unsqueeze(0)
-        boxes = boxes if boxes.is_floating_point() else boxes.float()
-
-        xmin, ymin = boxes[..., 0], boxes[..., 1]
-        mode = mode.lower()
-        if mode == "xyxy":
-            height, width = boxes[..., 3] - boxes[..., 1], boxes[..., 2] - boxes[..., 0]
-        elif mode == "xyxy_plus":
-            height, width = boxes[..., 3] - boxes[..., 1] + 1, boxes[..., 2] - boxes[..., 0] + 1
-        elif mode == "xywh":
-            height, width = boxes[..., 3], boxes[..., 2]
+        quadrilaterals: Union[torch.Tensor, List[torch.Tensor]]
```

Comment:
we must think of having a data structure Quadrilaterals too

Reply:
Yes. Out of scope for this PR, but we definitely need the data structure and apply it on all kornia functions